### PR TITLE
improve regression suite #96

### PR DESCRIPTION
**Pull request by spanners**
Add some targeted tests covering different Ada language features:

* testsuite/gnat2goto/tests/func_parameter_modes/func.adb
Add test for IN, OUT, and IN OUT parameter modes for functions.

* testsuite/gnat2goto/tests/if_expressions/if_expr.adb
Add test for if-expressions.

* testsuite/gnat2goto/tests/null_procedure/null_proc.adb
Add test for null procedures.

* testsuite/gnat2goto/tests/quantified_expressions/quant.adb
Add test for quantified expressions.

* testsuite/gnat2goto/tests/spec_expression_function/bar.ads
Add test for expression functions in specification files.

* testsuite/gnat2goto/tests/subtype_predicates/foo.adb
Add test for static predicates in subtype declarations.

### CODE REVIEW

File: testsuite/gnat2goto/tests/quantified_expressions/test.opt [hunk 1]
```diff
@@ -0,0 +1 @@
+ALL XFAIL gnat2goto crashes on quantified expression parsing
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/quantified_expressions/test.opt]: We will have to be very careful with this as it can be very difficult to support quantifiers but it should be possible to at least parse these and generate the appropriate irept.

File: testsuite/gnat2goto/tests/spec_expression_function/test.opt [hunk 2]
```diff
@@ -0,0 +1 @@
+ALL XFAIL cbmc range check is shadowed in module ''
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/spec_expression_function/test.opt]: Not sure about this...

File: testsuite/gnat2goto/tests/subtype_predicates/test.opt [hunk 3]
```diff
@@ -0,0 +1 @@
+ALL XFAIL gnat2goto crashes on static predicates
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/subtype_predicates/test.opt]: We will have to look at these, they may be difficult to support.

**Comment by martin-cs** [comment]
@thk123 / @chrisr-diffblue any chance of a quick review when you have a moment?

**Comment by martin-cs** [comment]
Replaced by #363 which is a rebased version of this.
